internal/app/machined/pkg/system/system.go

```diff
@@ -83,6 +83,34 @@ func (s *singleton) Load(services ...Service) []string {
 	return ids
 }
 
+// Reload adds service to the list of services managed by the runner.
+//
+// Reload returns service IDs for each of the services.
+func (s *singleton) Reload(services ...Service) []string {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.terminating {
+		return nil
+	}
+
+	ids := make([]string, 0, len(services))
+
+	for _, service := range services {
+		id := service.ID(s.runtime)
+		ids = append(ids, id)
+
+		if _, exists := s.state[id]; exists {
+			svcrunner := NewServiceRunner(service, s.runtime)
```

Comment:
idk, if we should do some check that service runner being replaced (contents of `s.state[id]`) is in some terminal state? as otherwise there might be a chance of duplicate service runners for the same service

Reply:
Good point.